[Teng Chiong] iP

tchiong opened the pull request:
### DukePro
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes/))

DukePro frees your mind of having to remember things you need to do. It's,
* text-based
* easy to learn
* ~~FAST~~ SUPER FAST to use

All you need to do is,
1. download it from [here](https://nus-cs2103-ay2122s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

tchiong committed:
Level-1

Implemented Skeleton of Duke

tchiong committed:
Level-2

Added Ability to store tasks into List and retrieve the List

tchiong committed:
Level-3

Added Task class,
Revamped List,
Added Checking task functionality

tchiong committed:
Level-4

Added 3 subclasses ToDo, Deadline, Event
Implemented Functionality

Could be improved

tchiong committed:
Level-5

Implemented Unique Error handling,

Rough and Rudimentary for now

tchiong committed:
Level-6

Added Delete functionality,

Added More error handling

tchiong committed:
A-TextUiTesting

Fleshed out the TextUiTesting to include errors,
Added Java Docs

tchiong committed:
Complete Level-7 requirements

Duke now stores list data locally @ localList.txt

Duke can retrieve and create a txt file to store list data in the same location as the application

Classes have been updated to include another constructor to accommodate creation from txt file data

greeting() function updated to read localList.txt if present and return a List<Task> which contains locally store list data

close() function now saves current List<Task> to localList.txt if present, if not it will create a localList.txt file and save it there

tchiong committed:
Complete Level-8

Firmed up Date formats for Deadline and Event, requiring user to use a specific format to input a Date

Changed logic to check for right date format, and pass a LocalDate into the constructors of Deadline and Event

Deadline and Event classes changed to take LocalDate instead of String, and return a custom format when displaying

tchiong committed:
Merge branch 'branch-Level-7'

tchiong committed:
Merged Branches Level-7 and Level-8

Merged the 2 branches Level-7 and Level-8

Code was cleaned up by fixing bugs in: - the deadline and event class
- localList.txt retrieval code in greeting()

tchiong committed:
Repack Code for MoreOOP

Code and logic was restructured into 4 new classes

DukeMain: Contains the main application and main function to run

DukeParser: Handles logic and understanding of user input

DukeUI: Handles User Interface related functions

DukeStorage: Handles Storage related functions to retrieve and save files

tchiong committed:
Package Classes

Sorted various classes into packages

tasks: Contains task class and subclasses

exception: Contains all created exceptions

Duke: Contains all Duke related functions and packages

tchiong committed:
Set-up JUnit

Set-up basic JUnit testing system

Tests DukeParser.parse() and DukeStorage.retrieve()

tchiong committed:
Create JAR

Created a JAR file for the application

tchiong committed:
Completed JavaDocs for current functions

Fleshed out JavaDocs for the functions in all the classes.

tchiong committed:
Completed Level-9

Implemented find [filter] that looks through the list of tasks to display tasks that contain filter in their descriptions

Added Functions in taskList, task to support

Added DukeUI functions to support

tchiong committed:
Check for CodingStandard Errors

Looked for Coding Standard Errors in the code

tchiong committed:
Merge branch 'branch-Level-9'

tchiong committed:
Merge branch 'branch-A-CodingStandard'

tchiong committed:
Merge branch 'add-gradle-support'

tchiong committed:
Add Gradle Support

Added Gradle Support for Duke

tchiong committed:
Implement Checkstyle

Implemented checkstyle and fixed the problems that were highlighted by checkstyle

radiankrisno commented on config/checkstyle/suppressions.xml:
```diff
@@ -0,0 +1,10 @@
+<suppressions>
+    <suppress checks="JavadocType" files=".*Test\.java"/>
+    <suppress checks="MissingJavadocMethodCheck" files=".*Test\.java"/>
+</suppressions>
```
Should end all files with a new line

radiankrisno commented on src/main/java/duke/DukeParser.java:
```diff
@@ -0,0 +1,207 @@
+
+    /**
+     * Class Constructor
+     * @param taskList The tasklist to be modified through user-input
```
Might want to give a new line between the method definition and @param tag

radiankrisno commented on src/main/java/duke/DukeStorage.java:
```diff
@@ -0,0 +1,83 @@
+import duke.tasks.Task;
+import duke.tasks.ToDo;
+
+public class DukeStorage {
```
You might want to add a header comment for the class

radiankrisno commented on src/test/java/duke/DukeStorageTest.java:
```diff
@@ -0,0 +1,29 @@
+
+public class DukeStorageTest {
+    @Test
+    public void test1() {
```
Maybe you want to follow test naming convention

radiankrisno commented on src/main/java/duke/tasks/Task.java:
```diff
@@ -0,0 +1,76 @@
+    public Task(String description, Type type, boolean done) {
+        this.type = type;
+        this.description = description;
+        this.isDone = done;
```
Do we need to use this keyword here?

tchiong replied:
Which keyword in particular are you referring to?

radiankrisno commented on src/main/java/duke/DukeParser.java:
```diff
@@ -0,0 +1,207 @@
+                time = LocalDate.parse(postFilter.get(1), standard);
+            }
+        } else {
+            //Error Handling (Can be improved)
```
I like how you add comments where improvements can be made!

radiankrisno commented on src/main/java/duke/classes/TaskList.java:
```diff
@@ -0,0 +1,95 @@
+
+import duke.tasks.Task;
+
+public class TaskList {
```
Might want to add header to other classes as well

radiankrisno commented on src/main/java/duke/tasks/Deadline.java:
```diff
@@ -0,0 +1,39 @@
+        return super.toString() + " (by: " + printOut.format(deadline) + ")";
+    }
+
+    public String toFileString() {
```
Might want to add javadocs for this method

radiankrisno reviewed (commented):
LGTM! The code is well written! Only minor changes are required to be made.

tchiong committed:
Completed Level-10

Created New UI using JavaFX

Linked UI to SceneBuilder

Revamped code to support new UI

DukeMain replaced with DukeGUI

DukeParser and DukeUI functions changed to return strings

rehmmann commented on src/main/java/duke/DukeStorage.java:
```diff
@@ -65,17 +68,16 @@ public DukeStorage(){
+    public void saveList(TaskList taskList) throws IOException {
+        File savedList = new File("localList.txt");
+        FileWriter listWriter = new FileWriter(savedList);
+        // Task format: [Type] [isDone] [Description] [Time/Deadline (if applicable)]
```
Perhaps remove commented out code if it is not needed anymore

rehmmann commented on src/test/java/duke/DukeParserTest.java:
```diff
@@ -33,26 +33,26 @@ public void Test1(){
         TaskList taskList = new TaskList(new ArrayList<>());
         DukeUI ui = new DukeUI();
 
         DukeParser test = new DukeParser(taskList, ui);
```
Could reduce whitespaces to improve readability

rehmmann commented on src/main/java/duke/DukeStorage.java:
```diff
@@ -65,17 +68,16 @@ public DukeStorage(){
-     * @param tasklist List to save to the file
+     * @param taskList List to save to the file
      * @throws IOException Thrown if there is an issue creating, retrieving or writing the file
      */
```
Good use of javadocs :)

rehmmann reviewed (commented):
Overall good job!It is readable and seems ready to be merged! Very few coding standard violations spotted

tchiong committed:
Added Assertions in Parser

Added Assertions in Parser to check for string input and legitimate key

tchiong committed:
Fixed coding violations and javadoc formatting

Fixed the coding violations and javadoc errors mentioned by code reviewers

tchiong committed:
Merge pull request #1 from tchiong/branch-A-Assertions

Added Assertions in Parser

tchiong committed:
Merge branch 'master' into branch-A-CodeQuality

tchiong committed:
Merge pull request #2 from tchiong/branch-A-CodeQuality

Fixed coding violations and javadoc formatting

tchiong committed:
Added help functionality

Added help to explain usage of other keywords, updated Keywords enumeration to work with help

tchiong committed:
A-BetterGui

Improvements to Gui in the form of a nicer background and duke profile picture

tchiong committed:
A-UserGuide

Completed the Read Me for Duke, adding all the features and descriptions for what the feature does, format to use and examples for the more complex ones

tchiong committed:
Minor Updates to class names

tchiong committed:
Refactor Classes and Clean Code

tchiong committed:
Refactor DukeParser

Refactored DukeParser into Parser and commands for better SLAP

tchiong committed:
Fix and Clean code

Fixed Bugs and Cleaned up unused imports and variables

tchiong committed:
Update README.md

tchiong committed:
Fix Bugs

Fixed Bugs in testing and check style errors
